src/point.c

```diff
@@ -263,3 +264,40 @@ spherepoint_equal(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(spoint_eq(p1, p2));
 }
+
+static SPoint * spherepoint_from_vector3d(Vector3D v){
+	SPoint* p = (SPoint *) palloc(sizeof(SPoint));
+	p->lat = asin(v.z / sqrt(pow(v.x, 2) + pow(v.y, 2)  + pow(v.z, 2)));
+	p->lng = atan2(v.y, v.x);
+	return p;
+}
+
+Datum get_gravity_center(PG_FUNCTION_ARGS) {
+	SPoint * p;
+	SPoint current_point;
+	Vector3D	v;
+	Vector3D	res;
+	ArrayType *dots_vector = PG_GETARG_ARRAYTYPE_P(0);
+	int num_elements = ArrayGetNItems(ARR_NDIM(dots_vector), ARR_DIMS(dots_vector));
+	SPoint *array_data = (SPoint *) ARR_DATA_PTR(dots_vector);
+
+	for (int i = 0; i < num_elements; i++) {
+		current_point = array_data[i];
+		//elog(LOG, "%lf\n", current_point.lat);
+		spoint_vector3d(&v, &current_point);
+		elog(LOG, "%lf\n", v.x);
+		res.x+=v.x;
+		res.y+=v.y;
+		res.z+=v.z;
```

Comment:
Add a space on both sides of the `+=` operator on lines 289, 290, and 291.

Reply:
Also, I think `res.x`, `res.y`, and `res.z` need to be initialized to zeroes before the `for` loop? This might explain the test failures seen on some PostgreSQL versions.

Reply:
What happens if `num_elements` is zero, i.e. if the ARRAY passed to the function is empty? I think it should give an error or return NULL perhaps?

Reply:
I think that the gravity center point (centroid) should be on the sphere in 2D space because pgsphere is about 2D space on the sphere. I'm not sure that the average point in 3d space will be on the sphere. May be to to average long/lat coordinates.